### PR TITLE
Show @main lenses with scala cli scripts

### PR DESCRIPTION
**Pull request by LaurenceWarne**
Hi!  This PR makes it so that `@main` lenses with scala cli scripts (e.g. files ending in `sc`) are shown.  They appeared previously only for scala cli files with the `.scala` extension, so the second added test would pass already, but the first wouldn't (though I'm unsure whether to put the new tests in `CodeLensLspSuite` or `CrossCodeLensLspSuite`).

I think this happened because the main (toplevel) function symbols for `sc` files typically have the form `_empty_/main.main().`, whereas `.scala` files would have the corresponding form `_empty_/main$package.main().`, and only the second form was accepted by `DebugProvider.dropSourceFromToplevelSymbol`.

Thanks

### CODE REVIEW

File: tests/unit/src/test/scala/tests/CodeLensLspSuite.scala [hunk 1]
```diff
@@ -211,6 +212,59 @@ class CodeLensLspSuite extends BaseCodeLensLspSuite("codeLenses") {
...
+        )
+      )
+      _ <- server.didOpen(path)
+      _ <- assertCodeLenses(
+        path,
+        """|<<run>><<debug>>
+           |val x = 3
+           |
+           |<<run>><<debug>>
+           |@main def main() = {
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/CodeLensLspSuite.scala:237]: I think this will only work for Scala 3, so better put it in the cross suite.

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for the contribution! This indeed works this way with main annotations in script files, let's just move the test to CrossCodeLensLspSuite

This will probably not show up for Scala 2 scripts as they will not compile at all.

**Comment by LaurenceWarne** [comment]
> let's just move the test to CrossCodeLensLspSuite

Done!

> This will probably not show up for Scala 2 scripts as they will not compile at all.

Out of interest, is there something equivalent to `@main` for scala-cli scripts under Scala 2 (afaik you can't use classes with a main method: https://github.com/VirtusLab/scala-cli/issues/1675) ?

**Comment by tgodzik** [comment]
> Out of interest, is there something equivalent to @main for scala-cli scripts under Scala 2 (afaik you can't use classes with a main method: https://github.com/VirtusLab/scala-cli/issues/1675) ?

I don't think there is anything like that. There might be a library doing something like that but not sure.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by LaurenceWarne** [comment]
Thanks!

**Pull request merged**